src/concat.ts

```diff
@@ -4,7 +4,11 @@ import { asUint8Array } from './util/as-uint8array.js'
 /**
  * Returns a new Uint8Array created by concatenating the passed ArrayLikes
  */
-export function concat (arrays: Array<ArrayLike<number>>, length?: number): Uint8Array {
+export function concat (arrays: Uint8Array[], length?: number): Uint8Array {
+  if (globalThis.Buffer != null) {
+    return asUint8Array(globalThis.Buffer.concat(arrays, length))
```

Comment:
Unfortunately `arrays` here causes Buffer.concat to `throw`, where before `arrays` would have been converted to a Uint8Array on line 24.